glyphs2fontir/src/source.rs

```diff
@@ -196,13 +197,49 @@ impl Work for StaticMetadataWork {
                     axis_values[idx][font.default_master_idx].into_inner() as f32,
                 );
 
-                // Given in user
-                let min = UserCoord::new(min);
-                let max = UserCoord::new(max);
-                let default = UserCoord::new(default);
-
-                // TODO process .glpyhs coordinate conversions
-                let converter = CoordConverter::unmapped(min, default, max);
+                // Given in design coords based on a sample file
+                let default = DesignCoord::new(default);
+                let min = DesignCoord::new(min);
+                let max = DesignCoord::new(max);
+
+                let converter = if font.axis_mappings.contains_key(&a.tag) {
+                    let mappings: Vec<_> = font
+                        .axis_mappings
+                        .get(&a.tag)
+                        .unwrap()
+                        .iter()
+                        .map(|(u, d)| (UserCoord::new(*u), DesignCoord::new(*d)))
+                        .collect();
+                    let default_idx = mappings
+                        .iter()
+                        .position(|(_, dc)| *dc == default)
+                        .unwrap_or_else(|| {
+                            panic!("Must have a mapping for default {:?} on {}", default, a.tag)
+                        });
+                    mappings
+                        .iter()
+                        .position(|(_, dc)| *dc == min)
+                        .unwrap_or_else(|| {
+                            panic!("Must have a mapping for min {:?} on {}", min, a.tag)
+                        });
+                    mappings
+                        .iter()
+                        .position(|(_, dc)| *dc == max)
+                        .unwrap_or_else(|| {
+                            panic!("Must have a mapping for max {:?} on {}", max, a.tag)
+                        });
+                    CoordConverter::new(mappings, default_idx)
+                } else {
+                    // There is no mapping; design == user
```

Comment:
when there is no Axis Mappings custom parameter, you can't simply assume design == user and go home..

You have to check for Axis Location custom parameters on the masters and optionally on the instances too, and use those to define {user-space: design-space} axis mappings. 

Have a read at "Axis mappings and locations" chapter of https://glyphsapp.com/learn/creating-a-variable-font

Actually, it's interesting that in that doc there's a box on the side with a warning like this:

> Axis Mappings are not always working as expected in 3.0.x, and may create faulty entries in the fvar table. We are considering deprecating this method. If possible, use the Axis Location method.

Well, glypysLib at least gives priority to the more general approach which is Axis Mappings, as the latter allows one to define more mappings than there are masters or instances (if one wished to warp the variation space with greater granularity)...

Finally, if neither Axis Mappings nor Axis Locations are defined.. well, you're in dangerous territory and glyphsLib will do something clever like taking the user-space locations implicitly assigner to instances (which do have weightClass and widthClass attribute corresponding to CSS values) and use those to infer the axis mappings.

Reply:
Axis Location's is TODO, I will update the comment to be more explicit about that

If neither is defined and you take the weight/width from attributes I would have thought you assume design == user?